File: src/main/java/me/miki/shindo/management/nanovg/font/Font.java
package me.miki.shindo.management.nanovg.font;

import lombok.Getter;
import lombok.Setter;
import net.minecraft.util.ResourceLocation;

import java.nio.ByteBuffer;

@Getter
public class Font {

    private final String name;
    private final ResourceLocation resourceLocation;

    @Setter
    private boolean loaded;

    @Setter
    private ByteBuffer buffer;

    public Font(String name, ResourceLocation resourceLocation) {
        this.name = name;
        this.resourceLocation = resourceLocation;
        this.loaded = false;
        this.buffer = null;
    }

}

File: src/main/kotlin/com/shindoclient/shindo/management/nanovg/font/Font.kt
package com.shindoclient.shindo.management.nanovg.font

import net.minecraft.util.ResourceLocation
import java.nio.ByteBuffer

class Font(
    val name: String,
    val resourceLocation: ResourceLocation,
    var isLoaded: Boolean = false,
    var buffer: ByteBuffer? = null,
)
